### PR TITLE
Support nixpkgs shallow clone and make use of github pull/<pr>/merge ref

### PR DESCRIPTION
**Pull request by azuwis**
Fix https://github.com/Mic92/nixpkgs-review/issues/267

### CODE REVIEW

File: nixpkgs_review/review.py [hunk 1]
```diff
@@ -272,15 +274,19 @@ def build_pr(self, pr_number: int) -> dict[System, list[Attr]]:
...
-            pr["base"]["ref"],
-            f"pull/{pr['number']}/head",
-        )
 
         if self.checkout == CheckoutOption.MERGE:
-            base_rev = merge_rev
+            base_rev, pr_rev = fetch_refs(
+                self.remote,
+                pr["base"]["ref"],
+                f"pull/{pr['number']}/merge",
```
Comment (Mic92) [inline nixpkgs_review/review.py:282]: In fact there is a race condition with this ref sometimes as github needs some time to compute it. But this is probably only relevant for CIs. So we should be unlikely to when humans run nixpkgs-review (except for @SuperSandro2000 maybe)

**Comment by Mic92** [comment]
@mergify queue

**Comment by azuwis** [comment]
I've found one regression, if the PR is merged, `pull/<PR>/merge` will be gone, and won't be able to run nixpkgs-review against a merged PR after this.

**Comment by Mic92** [comment]
Can we detect if `/pull/<PR>/merge` is available and otherwise fallback?

**Pull request merged**